dalston/block-extends/extend-cover-block.css

```diff
@@ -122,58 +167,82 @@
 	transition: opacity 0.2s ease-in-out;
 }
 
+.wp-block-cover.is-style-hover-reveal.dalston-in-viewport.has-background-dim.has-background-dim-0:before,
```

Comment:
As described in the issue, the `dim` styles are not working and I created an issue to check this.

Anyway, I added `.dalston-in-viewport` to all the `:hover` styles to keep the styles consistent. And when that issue is fixed, it should work.